<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>G4_43</class>
 <widget class="QWidget" name="G4_43">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>553</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>G4_43</string>
  </property>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>10</y>
     <width>661</width>
     <height>61</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'幼圆'; font-size:20pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;题目：设计父亲类&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;Father&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;、母亲类&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;Mother&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;和子女类&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;Child&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;，其主要数据成员是姓名、年龄和民族，子女继承了父亲的姓和母亲的民族。声明一个子女对象，并输出子女及其父母的姓名和民族信息。&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="father_age">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>266</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>28</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_15">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>270</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>年龄：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>222</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>民族：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>180</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>名字：</string>
   </property>
  </widget>
  <widget class="QPushButton" name="shang_jiao">
   <property name="geometry">
    <rect>
     <x>472</x>
     <y>330</y>
     <width>91</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>登  记</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="father_name">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>176</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>伞伞</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="father_nation">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>220</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>苗族</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>92</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>父  亲  信  息</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="father_first1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>128</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>张三</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>134</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>姓氏：</string>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>300</y>
     <width>211</width>
     <height>20</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>362</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>姓氏：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="mother_first">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>356</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>李四</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>462</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>民族：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="mother_name">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>408</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>司司</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="mother_nation1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>459</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>壮族</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="mother_age">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>510</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>28</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>412</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>名字：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_16">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>512</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>年龄：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>320</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>母  亲  信  息</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>130</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>姓氏：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="child_first">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>124</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>206</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>民族：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="child_name">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>164</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>伞司</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="child_nation">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>204</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="child_age">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>244</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>5</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>168</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>名字：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_17">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>248</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>年龄：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>88</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>子  女  信  息</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="Line" name="line_2">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>90</y>
     <width>20</width>
     <height>441</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
  </widget>
  <widget class="QListView" name="xian_shi_n">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>370</y>
     <width>401</width>
     <height>141</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="shang_jiao_2">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>330</y>
     <width>91</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>下 位 子 女</string>
   </property>
  </widget>
  <widget class="QLabel" name="ti_shi">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>520</y>
     <width>251</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_18">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>290</y>
     <width>41</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>性别：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="child_gender">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>288</y>
     <width>141</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>女</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>shang_jiao</sender>
   <signal>clicked()</signal>
   <receiver>G4_43</receiver>
   <slot>f_m_c_xian()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>486</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>530</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shang_jiao_2</sender>
   <signal>clicked()</signal>
   <receiver>G4_43</receiver>
   <slot>next_child()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>338</x>
     <y>301</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>father_first1</sender>
   <signal>textChanged(QString)</signal>
   <receiver>G4_43</receiver>
   <slot>father_first()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>192</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>237</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mother_nation1</sender>
   <signal>textChanged(QString)</signal>
   <receiver>G4_43</receiver>
   <slot>mother_nation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>196</x>
     <y>476</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>487</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>father_first()</slot>
  <slot>mother_nation()</slot>
  <slot>f_m_c_xian()</slot>
  <slot>next_child()</slot>
 </slots>
</ui>
